homeassistant/components/stiebel_eltron/climate.py

```diff
@@ -0,0 +1,153 @@
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    def update(self):
+        """Update unit attributes."""
+        self._ste_data.update()
+
+        self._target_temperature = self._ste_data.api.get_target_temp()
+        self._current_temperature = self._ste_data.api.get_current_temp()
+        self._current_humidity = self._ste_data.api.get_current_humidity()
+        self._filter_alarm = self._ste_data.api.get_filter_alarm_status()
+        self._current_operation = self._ste_data.api.get_operation()
+
+        _LOGGER.debug("Update %s, current temp: %s", self._name,
+                      self._current_temperature)
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        return {
+            ATTR_CURRENT_HUMIDITY: self._current_humidity,
+            'filter_alarm': self._filter_alarm
+        }
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._name
+
+    # Handle SUPPORT_TARGET_TEMPERATURE
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 0.1
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return 10.0
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return 30.0
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        if kwargs.get(ATTR_TEMPERATURE) is not None:
+            self._target_temperature = kwargs.get(ATTR_TEMPERATURE)
+        _LOGGER.debug("set_temperature: %s", self._target_temperature)
+        self._ste_data.api.set_target_temp(self._target_temperature)
+
+    # Handle SUPPORT_OPERATION_MODE
+    @property
+    def operation_list(self):
+        """List of the operation modes."""
+        return self._operation_modes
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return STE_TO_HA_STATE.get(self._current_operation)
+
+    def set_operation_mode(self, operation_mode):
+        """Set new operation mode."""
+        new_mode = HA_TO_STE_STATE.get(operation_mode)
+        _LOGGER.debug("set_operation_mode: %s -> %s", self._current_operation,
+                      new_mode)
+        self._current_operation = new_mode
```

Comment:
See above.

Reply:
Accordingly modified

Reply:
I would like to revert... same as above.

Reply:
I could modify the call to the update method to do a forced update. It's not an issue for the api.